src/hypothesis/strategies.py

```diff
@@ -174,16 +174,18 @@ def one_of(*args):
         except TypeError:
             pass
 
+    strategies = []
```

Comment:
This seems like it would result in the problem that you ran into while working on it that this ends up exposing a lot of internal strategy definitions in the repr.

It occurs to me though that an easy fix would be to add a defines\_strategy decorator to this function - the wrapper will sort out the repr automatically.